src/system_database.ts

```diff
@@ -636,18 +636,12 @@ export class PostgresSystemDatabase implements SystemDatabase {
   readonly workflowCancellationMap: Map<string, boolean> = new Map(); // Map from workflowID to its cancellation status.
 
   constructor(
-    readonly pgPoolConfig: PoolConfig,
-    readonly systemDatabaseName: string,
+    readonly systemDatabaseUrl: string,
     readonly logger: GlobalLogger,
     readonly sysDbPoolSize?: number,
   ) {
-    // Craft a db string from the app db string, replacing the database name:
-    const systemDbConnectionString = new URL(pgPoolConfig.connectionString!);
-    systemDbConnectionString.pathname = `/${systemDatabaseName}`;
-
     this.systemPoolConfig = {
-      connectionString: systemDbConnectionString.toString(),
-      connectionTimeoutMillis: pgPoolConfig.connectionTimeoutMillis,
```

Comment:
Knex needs `connectionTimeoutMillis` explicitly set: it won't pick it up from the connection string

Reply:
I don't think that's a knex issue. I don't think node pg picks up the connection timeout param out of the query parameters

Reply:
That's correct -- it is not knex specifically but rather PG -- for Knex and TypeORM, we need to pass it manually

Reply:
we need to pass it manually for all places we use the `pg` package. Will update

Reply:
added a helper function `getClientConfig` to set `connectionTimeoutMillis` manually